packages/dataviews/src/field-types/datetime.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * WordPress dependencies
+ */
+import { BaseControl } from '@wordpress/components';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	SortDirection,
+	ValidationContext,
+	DataFormControlProps,
+} from '../types';
+
+function sort( a: any, b: any, direction: SortDirection ) {
+	const timeA = new Date( a ).getTime();
+	const timeB = new Date( b ).getTime();
+
+	return direction === 'asc' ? timeA - timeB : timeB - timeA;
+}
+
+function isValid( value: any, context?: ValidationContext ) {
+	if ( context?.elements ) {
```

Comment:
If `elements` is provided, the date value should be one of those — otherwise is invalid. Note that the component renders as a SelectControl if the elements prop is present.

Reply:
But would we need `elements` in these types? Restricting dates usually is incorporated in the date picker components with a range of valid/invalid dates, no? No strong opinion here though..

Reply:
Yes, why wouldn't they? There's an example in the storybook for this that may help visualize a valid use case. It's poor UI (select), but think of "date presets" for example. In the future, we may even use this for showing the presets together with the time picker — but we're not there yet.